awslambda/src/test/scala/akka/stream/alpakka/awslambda/scaladsl/AwsLambdaFlowSpec.scala

```diff
@@ -0,0 +1,99 @@
+import org.mockito.ArgumentMatchers.{any => mockitoAny, eq => mockitoEq}
+import org.mockito.Mockito._
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mockito.MockitoSugar
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class AwsLambdaFlowSpec
+    extends TestKit(ActorSystem("AwsLambdaFlowSpec"))
+    with WordSpecLike
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with ScalaFutures
+    with Matchers
+    with MockitoSugar {
+
+  implicit val mat = ActorMaterializer()
+
+  implicit val ec = system.dispatcher
+
+  implicit val awsLambdaClient = mock[AWSLambdaAsyncClient]
+
+  override protected def afterEach(): Unit = {
+    reset(awsLambdaClient)
+    verifyNoMoreInteractions(awsLambdaClient)
+  }
+
+  override protected def afterAll(): Unit =
+    Await.ready(system.terminate(), 5.seconds)
+
+  "AwsLambdaFlow" should {
+
+    val invokeRequest = new InvokeRequest().withFunctionName("test_function").withPayload("test_payload")
+    val invokeFailureRequest = new InvokeRequest().withFunctionName("failure_function").withPayload("test_payload")
+    val invokeResult = new InvokeResult()
+    val lambdaFlow = AwsLambdaFlow(1)
+
+    "call a single invoke request" in {
+
+      when(awsLambdaClient.invokeAsync(mockitoEq(invokeRequest),
+          mockitoAny[AsyncHandler[InvokeRequest, InvokeResult]]())).thenAnswer(new Answer[AnyRef] {
+        override def answer(invocation: InvocationOnMock): AnyRef = {
+          invocation.getArgument[AsyncHandler[InvokeRequest, InvokeResult]](1).onSuccess(invokeRequest, invokeResult)
+          new CompletableFuture
+        }
+      })
+
+      val (probe, future) = TestSource.probe[InvokeRequest].via(lambdaFlow).toMat(Sink.seq)(Keep.both).run()
+      probe.sendNext(invokeRequest)
+      probe.sendComplete()
+
+      future.map(_ shouldBe invokeResult)
+      verify(awsLambdaClient, times(1)).invokeAsync(mockitoEq(invokeRequest),
+        mockitoAny[AsyncHandler[InvokeRequest, InvokeResult]]())
+
+    }
+
+    "call with exception" in {
+
+      when(awsLambdaClient.invokeAsync(mockitoAny[InvokeRequest](),
+          mockitoAny[AsyncHandler[InvokeRequest, InvokeResult]]())).thenAnswer(new Answer[Future[InvokeResult]] {
+        override def answer(invocation: InvocationOnMock): Future[InvokeResult] = {
+          invocation
+            .getArgument[AsyncHandler[InvokeRequest, InvokeResult]](1)
+            .onError(new RuntimeException("Error in lambda"))
+          new CompletableFuture[InvokeResult]()
+        }
+      })
+
+      val (probe, future) = TestSource.probe[InvokeRequest].via(lambdaFlow).toMat(Sink.seq)(Keep.both).run()
+
+      probe.sendNext(invokeFailureRequest)
+      probe.sendComplete()
+
+      future.failed.map(_ shouldBe a[RuntimeException])
+
+    }
```

Comment:
Thanks for pointing to this problem. 
I checked how aws_lambda_sdk is actually working and found that there is blocking io http client under the hood:
`pool-1-thread-19" #41 prio=5 os_prio=31 tid=0x00007fa0d0ce2000 nid=0x8503 runnable [0x0000700012e36000]
   java.lang.Thread.State: RUNNABLE
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:170)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	- locked <0x000000076f6b7450> (a java.lang.Object)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:930)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
	- locked <0x000000076f6b7040> (a sun.security.ssl.AppInputStream)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:139)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:155)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:284)
...
`
So probably I need to make a calls to AWS Lambda using non-blocking client, like it was implemented in akka.stream.alpakka.s3.impl.S3Stream. 
Any other suggestions? 

Thanks :)